File: Services/pkg/models/filter.go
package models

type Search struct {
	Keyword string `query:"keyword" json:"keyword"`
	Column  string `query:"column" json:"column"`
	Text    string `query:"text" json:"text"` // For Semantic Search
}

func (s *Search) GetSearchString() string {
	return "keyword=" + s.Keyword + "&column=" + s.Column
}

File: hono-clean-architecture/pkg/models/filter.ts
export interface Search {
  keyword?: string;
  column?: string;
  text?: string;
}

export function createSearch(keyword?: string, column?: string, text?: string): Search {
  return {
    keyword: keyword || undefined,
    column: column || undefined,
    text: text || undefined,
  };
}

export function getSearchString(search: Search): string {
  return `keyword=${search.keyword || ""}&column=${search.column || ""}&text=${search.text || ""}`;
}
